### PR TITLE
:alien: add support for aws_autoscaling_group filters

### LINKED ISSUE #3534: Improve filters for data source aws_autoscaling_group
When using aws_autoscaling_group as a data source, I expect to be able to filter autoscaling groups similar to how I can filter other resource (i.e. aws_ami). 

### Terraform Version
Any

### Affected Resource(s)
aws_autoscaling_groups 

### Terraform Configuration Files
```hcl
data "aws_autoscaling_groups" "asgs" {
  filter {
    name = "tag:some-arbitrary-tag"
    values = ["some-arbitrary-value", "some-other-arbitrary-value"]
  }
}
```

### Expected Behavior
Finds AutoScaling groups that have a tag with a key `some-arbitrary-value` and either `some-arbitrary-value` or `some-other-arbitrary-value` as values.

### Actual Behavior
This behavior isn't supported.

### Preliminary Findings
I spent a little time seeing if I could implement this behavior myself. I can see why this isn't currently supported :). The data provider (`terraform-provider-aws/aws/data_source_aws_autoscaling_groups.go`) basically just encapsulates the behavior of the AWS SDK's `DescribeTags()` functionality, which means this data source isn't doing what we (users) would intuitively expect. I see two possible routes to fix this:
1. Increase the complexity of the data source by implementing the desired behavior 
2. Cooperate with Amazon to improve the quality of the SDK in this instance

Option 1 is probably the way to solve the issue fastest, but the real problem is that the SDK doesn't easily support the desired behavior, so option 2 is probably the correct solution. Perhaps we implement a temporary workaround, while looking at how we can improve the SDK to better suit this behavior?

Anyway, I'd be willing to help on this issue of the team provides some feedback on the proper approach.

### PR DESCRIPTION
**Pull request by danquack**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Some people have asked me on the 👽 . These emojis are defined by gitmoji.dev, which relates to `Update code due to external API changes.`

Relates OR Closes #3534

Output from acceptance testing:

```
~ make testacc TESTARGS="-run=TestAccAutoScalingGroupsDataSource_basic" PKG_NAME="./internal/service/autoscaling" ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ././internal/service/autoscaling/... -v -count 1 -parallel 3 -run=TestAccAutoScalingGroupsDataSource_basic -timeout 180m
=== RUN   TestAccAutoScalingGroupsDataSource_basic
=== PAUSE TestAccAutoScalingGroupsDataSource_basic
=== CONT  TestAccAutoScalingGroupsDataSource_basic
--- PASS: TestAccAutoScalingGroupsDataSource_basic (75.92s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/autoscaling        79.298s
```

### CODE REVIEW

File: internal/service/autoscaling/groups_data_source_test.go [hunk 1]
```diff
@@ -24,11 +24,20 @@ func TestAccAutoScalingGroupsDataSource_basic(t *testing.T) {
 				Config: testAccCheckGroupsConfig(sdkacctest.RandInt(), sdkacctest.RandInt(), sdkacctest.RandInt()),
 			},
 			{
-				Config: testAccCheckGroupsWithDataSourceConfig(sdkacctest.RandInt(), sdkacctest.RandInt(), sdkacctest.RandInt()),
```
Comment (danquack) [inline internal/service/autoscaling/groups_data_source_test.go:27]: The resources were not being created with this function, but rather using the `testAccCheckGroupsConfig` above resources. Therefore, I removed some of this cruft.

File: internal/service/autoscaling/filters.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package autoscaling
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func buildFiltersDataSource(set *schema.Set) []*autoscaling.Filter {
```
Comment (drewmullen) [inline internal/service/autoscaling/filters.go:9]: 🎨  unless the functions are being reused, we prefer to keep them as close to the relevant files as possible. these can be moved directly into the data_source file

**Review by drewmullen — CHANGES_REQUESTED** [review]
great work! thanks for your contribution. 1 🎨  stylistic change requested

**Review by drewmullen — APPROVED** [review]
lgtm! 🚀

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS="-run=TestAccAutoScalingGroupsDataSource_basic" PKG_NAME="./internal/service/autoscaling"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ././internal/service/autoscaling/... -v -count 1 -parallel 20 -run=TestAccAutoScalingGroupsDataSource_basic -timeout 180m
=== RUN   TestAccAutoScalingGroupsDataSource_basic
=== PAUSE TestAccAutoScalingGroupsDataSource_basic
=== CONT  TestAccAutoScalingGroupsDataSource_basic
--- PASS: TestAccAutoScalingGroupsDataSource_basic (47.78s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/autoscaling	51.526s
```

**Comment by ewbankkit** [comment]
@danquack Thanks for the contribution.
In general the changes look great. I went ahead and simplified the acceptance testing configuration(s) even more, consolidating into a single generating function (the fact that the data sources weren't referencing or depending on any of the autoscaling group resources may lead to intermittent tests failures).

**Pull request merged**